### PR TITLE
RANGER-4330: Fix ranger container startup due to arch change in ubuntu base image

### PR DESCRIPTION
**Pull request by kumaab**
## What changes were proposed in this pull request?
There is a change in CPU arch from amd64 to arm64 in the ubuntu 20.04 base image, the Java path changes due to this which affects all docker containers since all images depend on this base image.

### Error logs
```
2023-07-25 12:16:50 2023-07-25 19:16:50,735   --------- Running Ranger PolicyManager Web Application Install Script --------- 
2023-07-25 12:16:50 2023-07-25 19:16:50,738  [I] uname=Linux
2023-07-25 12:16:50 2023-07-25 19:16:50,741  [I] hostname=ranger.example.com
2023-07-25 12:16:50 2023-07-25 19:16:50,746  [I] DB_FLAVOR=POSTGRES
2023-07-25 12:16:50 2023-07-25 19:16:50,750  [I] Audit source=solr
2023-07-25 12:16:50 2023-07-25 19:16:50,754  [I] Checking distribution name..
2023-07-25 12:16:50 2023-07-25 19:16:50,789  [I] Found distribution : Ubuntu
2023-07-25 12:16:50 2023-07-25 19:16:50,790  [I] check if command /usr/lib/jvm/java-8-openjdk-amd64/bin/java exists
2023-07-25 12:16:50 ./setup.sh: line 195: type: /usr/lib/jvm/java-8-openjdk-amd64/bin/java: not found
2023-07-25 12:16:50 2023-07-25 19:16:50,792  [E] '/usr/lib/jvm/java-8-openjdk-amd64/bin/java' command not found
2023-07-25 12:16:50 find: '/opt/ranger/ranger-2.4.1-admin/ews/webapp/WEB-INF/classes/conf/': No such file or directory
2023-07-25 12:16:50 Starting Apache Ranger Admin Service
2023-07-25 12:16:50 ./ews/ranger-admin-services.sh: line 94: /opt/ranger/ranger-2.4.1-admin/ews/logs/catalina.out: No such file or directory
2023-07-25 12:16:55 Apache Ranger Admin Service failed to start!
```

## How was this patch tested?

Tested changes by bringing up ranger in docker and verifying ranger-admin is working.
### New Logs
```
2023-07-25 15:42:57 2023-07-25 22:42:57,994   --------- Running Ranger PolicyManager Web Application Install Script --------- 
2023-07-25 15:42:57 2023-07-25 22:42:57,996  [I] uname=Linux
2023-07-25 15:42:57 2023-07-25 22:42:57,999  [I] hostname=ranger.example.com
2023-07-25 15:42:58 2023-07-25 22:42:58,004  [I] DB_FLAVOR=POSTGRES
2023-07-25 15:42:58 2023-07-25 22:42:58,007  [I] Audit source=solr
2023-07-25 15:42:58 2023-07-25 22:42:58,011  [I] Checking distribution name..
2023-07-25 15:42:58 2023-07-25 22:42:58,049  [I] Found distribution : Ubuntu
2023-07-25 15:42:58 2023-07-25 22:42:58,050  [I] check if command /usr/lib/jvm/java-8-openjdk-arm64/bin/java exists
2023-07-25 15:42:58 2023-07-25 22:42:58,052  [I] '/usr/lib/jvm/java-8-openjdk-arm64/bin/java' command found
2023-07-25 15:42:58 2023-07-25 22:42:58,123  [I] Checking POSTGRES CONNECTOR FILE : /usr/share/java/postgresql.jar
2023-07-25 15:42:58 2023-07-25 22:42:58,125  [I] POSTGRES CONNECTOR FILE : /usr/share/java/postgresql.jar file found
2023-07-25 15:42:58 2023-07-25 22:42:58,126  [I] Setting up UNIX user : ranger and group: ranger
2023-07-25 15:42:58 2023-07-25 22:42:58,133  [I] the ranger user already exists and belongs to group ranger
2023-07-25 15:42:58 2023-07-25 22:42:58,134  [I] Setting up UNIX user : ranger and group: ranger DONE
2023-07-25 15:42:58 2023-07-25 22:42:58,136  [I] Setting up installation files and directory
2023-07-25 15:42:58 2023-07-25 22:42:58,137  [I] Copying /opt/ranger/admin/ews/webapp/WEB-INF/classes/conf.dist /opt/ranger/admin/ews/webapp/WEB-INF/classes/conf
2023-07-25 15:42:58 2023-07-25 22:42:58,143  [WARN] Property hadoop_conf not found. Creating blank core-site.xml.
2023-07-25 15:42:58 2023-07-25 22:42:58,145  [I] Creating /opt/ranger/admin/ews/webapp/WEB-INF/classes/lib
```

### CODE REVIEW

File: dev-support/ranger-docker/Dockerfile.ranger-base [hunk 1]
```diff
@@ -25,7 +25,7 @@ RUN apt-get update && \
     pip3 install requests
 
 # Set environment variables
-ENV JAVA_HOME      /usr/lib/jvm/java-8-openjdk-amd64
+ENV JAVA_HOME      /usr/lib/jvm/java-8-openjdk-arm64
```
Comment (mneethiraj) [inline dev-support/ranger-docker/Dockerfile.ranger-base]: RANGER-4181 replaced hardcoding of "amd64" with TARGETARCH. I suggest to consider a similar approach here.

Comment (kumaab) [reply dev-support/ranger-docker/Dockerfile.ranger-base]: Sure, I've updated the patch.

File: dev-support/ranger-docker/README.md [hunk 2]
```diff
@@ -40,6 +40,7 @@ Docker files in this folder create docker images and run them to build Apache Ra
    export DOCKER_BUILDKIT=1
    export COMPOSE_DOCKER_CLI_BUILD=1
    export RANGER_DB_TYPE=postgres
+   export TARGETARCH=arm64
```
Comment (mneethiraj) [inline dev-support/ranger-docker/README.md]: @kumaab  - TARGETARCH will be automatically set if following env is set while building the image:

`DOCKER_BUILDKIT=1 `

More on this at https://docs.docker.com/build/buildkit/ and https://docs.docker.com/engine/reference/builder/.

Comment (kumaab) [reply dev-support/ranger-docker/README.md]: Got it, thank you!

**Comment by kumaab** [comment]
Verified ranger container running fine with the latest changes on the PR.

**Review by mneethiraj — APPROVED** [review]

**Pull request merged**